docs/source/plugin/advanced.rst

```diff
@@ -15,6 +15,59 @@ If something is not in here, feel free to ask about it on our IRC channel, or
 maybe open an issue with the solution if you devise one yourself.
 
 
+Restricting commands to certain channels
+========================================
+
+Allowing games, for example, to be run only in specific channels is a
+relatively common request, but a difficult feature to support directly in
+Sopel's plugin API. Fortunately it is fairly trivial to build a custom
+decorator function that handles this in a configurable way.
+
+Here is a sample plugin that defines such a custom decorator, along with the
+scaffolding needed to pull a list of channels from the bot's settings::
+
+    import functools
+
+    from sopel import plugin
+    from sopel.config import types
+
+
+    class MyPluginSection(types.StaticSection):
+        allowed_channels = types.ListAttribute('allowed_channels', default=['#botspam'])
+
+
+    def setup(bot):
+        bot.config.define_section('myplugin', MyPluginSection)
+
+
+    def my_plugin_require_channel(func):
+        @functools.wraps(func)
+        def decorated(bot, trigger):
+            if trigger.sender not in bot.settings.myplugin.allowed_channels:
+                return
+            return func(bot, trigger)
+        return decorated
+
+
+    @plugin.command('command_name')
+    @plugin.require_chanmsg
+    @my_plugin_require_channel
+    def my_command(bot, trigger):
+        bot.say('This is the good channel.')
+
+.. important::
+
+    When using this example in your own plugin code, remember to change
+    ``myplugin`` to a section name appropriate for your plugin. It is also a
+    good idea to rename the ``MyPluginSection`` class accordingly.
+
+.. note::
+
+    This kind of decorator could be written in many different ways; this
+    example is just the simplest. Implementation of more complex approaches
```

Comment:
I'm always reluctant to use words like "simplest" or "easiest". In that case, I'd say "is the most common usage". This would reflect what people ask on IRC and what we tell them to do, so that count as "most common" IMHO.

Reply:
Pushed some more edits, including to fix the "old habits die hard" above w/`bot.config`.